pkg/core/config/tenant_test.go

```diff
@@ -521,4 +521,20 @@ user_config:
 			So(string(bodyBytes), ShouldEqualJSON, inputMinimalJSON)
 		})
 	})
+
+	Convey("Test updateNilFieldsWithZeroValue", t, func() {
```

Comment:
Can you add one more test case for the recursive behavior?

Reply:
Updated the test! @kiootic 

Btw, I found that the function may fail to exit, if the calling struct has the same type of pointer in the same path (e.g. self referencing). I added one more test to check the `UserConfiguration` to ensure it doesn't fall into the case.